src/main/java/org/alfresco/repo/solr/SOLRTrackingComponentImpl.java

```diff
@@ -865,7 +865,26 @@ public void getNodesMetadata(
                 }
                 else
                 {
-                   throw new AlfrescoRuntimeException("Nodes with no type are ignored by SOLR");
+                    QName typeQName = null;
+                    TypeDefinition typeDefinition = null;
+                    
+                    String errorMessage = "NodeId " + nodeId + " with nodeRef " + nodeRef;
+
+                    typeQName = nodeDAO.getNodeType(nodeId);
+                    if (typeQName != null)
+                    {
+                        typeDefinition = dictionaryService.getType(typeQName);
+                        if (typeDefinition == null)
+                        {
+                            errorMessage += " has type " + typeQName + ", but this type is not registered in DictionaryService.";
+                        }

```

Comment:
Missing an "else" here.

Reply:
Thanks, I've simplified the code at this point.